index.js

```diff
@@ -99,20 +99,26 @@ const init = ({
     }
     let shouldSkipTest;
 
-    beforeEach(() => {
+    const beforeCallback = () => {
       shouldSkipTest = canSkipTest();
       if (shouldSkipTest) return
 
       console[methodName] = newMethod // eslint-disable-line no-console
       unexpectedConsoleCallStacks.length = 0
```

Comment:
For this change to work as expected, I think that we need to check for the unexpected console callstacks corresponding to the beforeAll in the first beforeEach. This needs to be tested